deps/src/polymake.cpp

```diff
@@ -95,32 +104,43 @@ JLCXX_MODULE define_module_polymake(jlcxx::Module& polymake)
           return show_small_object<WrappedT>(V);
         });
     });
-  POLYMAKE_INSERT_TYPE_IN_MAP_SINGLE_TEMPLATE(pm_Vector,pm_Integer);
-  POLYMAKE_INSERT_TYPE_IN_MAP_SINGLE_TEMPLATE(pm_Vector,pm_Rational);
+  polymake.method("to_vector_Integer", [](pm::perl::PropertyValue pv){
+    return to_SmallObject<pm::Vector<pm::Integer>>(pv);
+  });
+  polymake.method("to_vector_Rational", [](pm::perl::PropertyValue pv){
+    return to_SmallObject<pm::Vector<pm::Rational>>(pv);
+  });
+    
+  POLYMAKE_INSERT_TYPE_IN_MAP(pm_Vector_pm_Integer);
+  POLYMAKE_INSERT_TYPE_IN_MAP(pm_Vector_pm_Rational);
 
   polymake.method("initialize_polymake", &initialize_polymake);
   polymake.method("call_func_0args",&call_func_0args);
   polymake.method("call_func_1args",&call_func_1args);
   polymake.method("call_func_2args",&call_func_2args);
-  polymake.method("application",[](const std::string x){ data.main_polymake_session->set_application(x); });
+  polymake.method("application",[](const std::string x){
+    data.main_polymake_session->set_application(x);
+  });
 
+  polymake.method("to_bool",[](pm::perl::PropertyValue p){ return static_cast<bool>(p);});
   polymake.method("to_int",[](pm::perl::PropertyValue p){ return static_cast<int64_t>(p);});
   polymake.method("to_double",[](pm::perl::PropertyValue p){ return static_cast<double>(p);});
-  polymake.method("to_bool",[](pm::perl::PropertyValue p){ return static_cast<bool>(p);});
   polymake.method("to_perl_object",&to_perl_object);
-  polymake.method("to_pm_Integer",&to_pm_Integer);
-  polymake.method("to_pm_Rational",&to_pm_Rational);
-  polymake.method("to_vector_rational",to_vector_rational);
-  polymake.method("to_vector_int",to_vector_integer);
-  polymake.method("to_matrix_rational",to_matrix_rational);
-  polymake.method("to_matrix_int",to_matrix_integer);
 
-  polymake.method("typeinfo_string", [](pm::perl::PropertyValue p){ PropertyValueHelper ph(p); return ph.get_typename(); });
-  polymake.method("check_defined",[]( pm::perl::PropertyValue v){ return PropertyValueHelper(v).check_defined();});
+  polymake.method("typeinfo_string", [](pm::perl::PropertyValue p){
+    PropertyValueHelper ph(p); 
+    return ph.get_typename();
+  });
 
   polymake_module_add_set(polymake);
   POLYMAKE_INSERT_TYPE_IN_MAP(pm_Set_Int64);
   POLYMAKE_INSERT_TYPE_IN_MAP(pm_Set_Int32);
+  
+  polymake_module_add_array(polymake);
+  POLYMAKE_INSERT_TYPE_IN_MAP(pm_Array_Int32);
```

Comment:
I am not sure what should be the best practice, actually. If you include `polymake_caller.h` in the `polymake_array.cpp`, you could also call this macro in the polymake array file. Which version do you prefer?

Reply:
ok, should we always call `POLYMKE_INSERT_TYPE_IN_MAP` inside the appropriate cpp files?

Reply:
I am not sure. It would probably be make more sense there, but it would also be more distributed and therefore easier to miss.

Reply:
Maybe it's best to have it where the type is defined? @saschatimme what do You think?